docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
+
+The table below outlines the relationship between Discord and Python types.
+
++-----------------+------------------------------------------------------------------------------------+
+|   Discord Type  |                                Python Type                                         |
++=================+====================================================================================+
+| ``STRING``      | :class:`str`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``INTEGER``     | :class:`int`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``BOOLEAN``     | :class:`bool`                                                                      |
++-----------------+------------------------------------------------------------------------------------+
+| ``NUMBER``      | :class:`float`                                                                     |
++-----------------+------------------------------------------------------------------------------------+
+| ``USER``        | :class:`~discord.User` or :class:`~discord.Member`                                 |
++-----------------+------------------------------------------------------------------------------------+
+| ``CHANNEL``     | :class:`~discord.abc.GuildChannel` and all subclasses, or :class:`~discord.Thread` |
++-----------------+------------------------------------------------------------------------------------+
+| ``ROLE``        | :class:`~discord.Role`                                                             |
++-----------------+------------------------------------------------------------------------------------+
+| ``MENTIONABLE`` | :class:`~discord.User` or :class:`~discord.Member`, or :class:`~discord.Role`      |
++-----------------+------------------------------------------------------------------------------------+
+| ``ATTACHMENT``  | :class:`~discord.Attachment`                                                       |
++-----------------+------------------------------------------------------------------------------------+
+
+:ddocs:`Application command option types <interactions/application-commands#application-command-object-application-command-option-type>` as documented by Discord.
+
+User parameter
+^^^^^^^^^^^^^^^
+
+Annotating to either :class:`discord.User` or :class:`discord.Member` both point to a ``USER`` Discord-type.
+
+The actual type given by Discord is dependent on whether the command was invoked in DM-messages or in a guild.
+
+For example, if a parameter annotates to :class:`~discord.Member`, and the command is invoked in a guild,
+discord.py will raise an error since the actual type given by Discord,
+:class:`~discord.User`, is incompatible with :class:`~discord.Member`.
```

Comment:
```suggestion
For example, if a parameter annotates to :class:`~discord.Member`, and the command is invoked in a DM,
discord.py will raise an error since the actual type given by Discord,
:class:`~discord.User`, is incompatible with :class:`~discord.Member`.
```